### DISCUSSION TITLE
Error ellipses in Kalman Filtered Argos data for mp SSM

### DISCUSSION THREAD

**Discussion by AdriGooden** [ianjonsen/foieGras#58] [Q&A] [answered]
Hello 

I was wondering what the impact would be on the 'mp' SSM model if we are unable to include the semi-major/minor and error ellipses for Kalman-filtered Argos data. I had a bit of an issue with the formatting of some of the datetimes on my full csv and may need to use a smaller csv which doesn't have the datetime issue but is also lacking the ellipse information. So I was wondering what removing this data would mean for the model. 

Could the semi-major/minor and error ellipses and their role in the model please be explained more fully? 

I really appreciate your help with this matter. 

Thank you

Adrienne

**Comment by ianjonsen** [comment, marked as answer]
Hi Adrienne, 

First, thank you for putting this question in the Discussions section, rather than Issues! 

We use these estimates of location uncertainty in the SSM's observation equation to account for the measurement error in the Argos locations. The error ellipse information provided with Argos Kalman-filtered locations gives more detailed information about the uncertainty associated with the Argos KF locations than we would get just going by the location quality class (`lc`). 

You can fit SSM's to Argos KF data that lack the error ellipse information. In this case, the KF-derived locations will be treated as though they are the older Argos Least-Squares locations - eg. we use the location quality class along with independent data built into AniMotum to parameterize the SSM's observation model. This may result in SSM-predicted locations that are a bit less precise than they could be if you had the error ellipse info, but the differences should usually be small.

Ian

**Comment by AdriGooden** [comment]
Thanks Ian for answering my questions and providing that information!

**Comment by AdriGooden** [comment]
Hi Ian

I have a follow on question in regards to the error ellipse data. I was able to sort my datetime issue (it was an excel problem) and can include the error ellipse data now. As such I am adding datetime and location for the release and recaptures of my sharks, but these data points are missing the eor, smaj and smin data as they are actual location taken from the boat. 

On the aniMotum vignettes overview (https://ianjonsen.github.io/aniMotum/articles/Overview.html) it states that Missing error ellipse values are allowed. What is the correct way to report this data? I have tried NAs for these points but have been getting an error message so I am guessing I am not reporting this correctly as the model was running fine prior to this. 
`Error in if (p.GL > 0.75) { : missing value where TRUE/FALSE needed`

I appreciate your help with this

Thanks 

Adrienne

**Comment by ianjonsen** [comment]
The answer depends on how your data, as input to `fit_ssm()`, is structured. If you're using a simple data.frame then you need to make sure that those first and last locations have `lc = "G"`, assuming the boat locations are measured by GPS. The `smaj`, `smin`, and `eor` variables can be `NA` for those two locations. 

If the above doesn't solve your problem then you will have to attach you data to your reply so I can figure out what is happening.

Cheers, Ian

**Comment by AdriGooden** [comment]
Thanks for your help Ian that was the answer! 

Cheers

Adrienne
